router/core/context.go

```diff
@@ -137,6 +138,96 @@ var metricAttrsPool = sync.Pool{
 	},
 }
 
+// The mapper can include elements, which are not part of the context, but it should be possible
+// to also configure them in the same way as the context fields.
+const (
+	filterKeyOperationName       = ContextFieldOperationName
+	filterKeyOperationHash       = ContextFieldOperationHash
+	filterKeyRouterConfigVersion = "router_config_version"
+)
+
+type attributeKeyMap map[attribute.Key]string
+
+// attributeMapper resolves attributes from the requestContext based on the configured custom attributes
+type attributeMapper struct {
+	enabled bool
+
+	// lookupMap contains the default key names, which can be mapped to a new name with custom attributes.
+	// These keys are used to identify attributes, which should not be included by default and map to the
+	// corresponding fields in the request context.
+	lookupMap attributeKeyMap
+	// attr is the list of configured custom context attributes. This list indicates whether attributes are
+	// added to the metrics and potentially remapped to a new key. Only attributes from the lookupMap are taken
+	// into account.
+	attr []config.CustomAttribute
+}
+
+func newAttributeMapper(enabled bool, attr []config.CustomAttribute) attributeMapper {
+	// Any attributes that are in this map, will be resolved only if they are configured in the `attr` list.
+	// This is to avoid adding certain attributes which might cause higher cardinality to the metrics
+	// by default as it can be expensive for the metric backend.
+	set := attributeKeyMap{
+		otel.WgOperationName:       filterKeyOperationName,
+		otel.WgOperationHash:       filterKeyOperationHash,
+		otel.WgRouterConfigVersion: filterKeyRouterConfigVersion,
+	}
+
+	return attributeMapper{
+		enabled:   enabled,
+		attr:      attr,
+		lookupMap: set,
+	}
+}
+
+func (r *attributeMapper) mapAttributes(attributes []attribute.KeyValue) []attribute.KeyValue {
+	if !r.enabled {
```

Comment:
I tested it locally and `enabled` is always false. Shouldn't it be called whenever I want to add a custom metric attribute?

```yaml
telemetry:
  metrics:
    attributes:
      - key: a
        value_from:
          context_field: operation_name
```

Reply:
I understood that this is the correct behavior? If the default cloud exporter is set, we do not want to filter out or remap any attributes and keep the attributes in.